File: articles/geography-coordinates-and-computational-mapping-in-the-islamic-world/c/route_distance.c
#include <stdio.h>

int main(void) {
    double segments[] = {12.0, 20.0, 7.5};
    double total = 0.0;
    for (int i = 0; i < 3; i++) total += segments[i];
    printf("segments=%d\n", 3);
    printf("total_distance=%.6f\n", total);
    return 0;
}

File: articles/geography-coordinates-and-computational-mapping-in-the-islamic-world/java/RouteDistance.java
public class RouteDistance {
    public static void main(String[] args) {
        double[] segments = {12.0, 20.0, 7.5};
        double total = 0.0;
        for (double s : segments) total += s;
        System.out.printf("segments=%d%n", segments.length);
        System.out.printf("total_distance=%.6f%n", total);
    }
}
